tools/codegen/diff_native_functions.py

```diff
@@ -0,0 +1,143 @@
+import sys
+import tempfile
+
+# TODO(jwtan): Figure out why we need this.
+root = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+sys.path.insert(0, root)
+
+from tools.codegen.gen import parse_native_yaml
+
+# Returns the current_version and the base_version if specified by users.
+def get_versions() -> (str, str):
+    # TODO(jwtan): It returns a lot of information including release notes. Figure out a way to only query the tag names.
+    # TODO(jwtan): per_page=100 is not future proof.
+    response = requests.get("https://api.github.com/repos/pytorch/pytorch/releases?per_page=100")
+    releases = [release['tag_name'] for release in response.json()]
+
+    current_version = releases[0]
+
+    parser = argparse.ArgumentParser(description='Diff native functions. Please run this script in the open-source repository.')
+    parser.add_argument(
+        '-c',
+        '--current-version',
+        help='current PyTorch version, default: ' + current_version,
+        default=current_version)
+    parser.add_argument(
+        '-b',
+        '--base-version',
+        help='base PyTorch version for comparison, default to the previous version')
+    options = parser.parse_args()
+
+    current_version = options.current_version
+    base_version = options.base_version
+
+    release_map = {value: index for index, value in enumerate(releases)}
+    if (current_version not in release_map.keys()) or ((base_version is not None) and (base_version not in release_map.keys())):
+        # TODO(jwtan): Figure out the conventional way to log/print error.
+        print('Input versions are invalid. Valid versions are {}.'.format(releases))
+
+    if base_version is None:
+        base_version = releases[release_map[current_version] + 1]
+
+    if release_map[current_version] >= release_map[base_version]:
+        # TODO(jwtan): Figure out the conventional way to log/print error.
+        print('Base version should be older than the current version.')
+
+    return current_version, base_version
+
+# Returns the corresponding commits for current_version and the base_version.
+# TODO(jwtan): Can we combine get_versions() and get_commits(...)?
+def get_commits(current_version: str, base_version: str) -> (str, str):
+    # TODO(jwtan): per_page=100 is not future proof.
+    response = requests.get("https://api.github.com/repos/pytorch/pytorch/tags?per_page=100")
+
+    current_commit_hash = ''
+    base_commit_hash = ''
+    for tag in response.json():
+        if tag['name'] == current_version:
+            current_commit_hash = tag['commit']['sha']
+        if tag['name'] == base_version:
+            base_commit_hash = tag['commit']['sha']
+
+    return current_commit_hash, base_commit_hash
+
+# Writes the native_functions.yaml of the provided commit to a temp file, and returns the absolute path to the file.
+# Caller should unlink the files when done.
+def get_yaml(commit_hash: str):
+    handle, path = tempfile.mkstemp()
```

Comment:
Use TemporaryFile instead of mkstemp, it is more idiomatic and ensures cleanup happens even if the subprocess invocation fails